theano/sandbox/cuda/opt.py

```diff
@@ -741,11 +749,11 @@ def local_gpu_gemm(node):
         y_on_gpu = (y.owner and isinstance(y.owner.op, HostFromGpu))
         z_on_gpu = (z.owner and isinstance(z.owner.op, HostFromGpu))
         if x_on_gpu or y_on_gpu or z_on_gpu:
-            return [host_from_gpu(gpu_gemm_no_inplace(as_cuda_ndarray_variable(z),
-                                                 a,
-                                                 as_cuda_ndarray_variable(x),
-                                                 as_cuda_ndarray_variable(y),
-                                                 b))]
+            return [host_from_gpu(gpu_gemm_no_inplace(gpu_from_host(z),
+                                                      a,
+                                                      gpu_from_host(x),
+                                                      gpu_from_host(y),
```

Comment:
This is a bad change.  as_cuda_ndarray_variable() will work on stuff that is already on the gpu whereas gpu_from_host() will not.

Also why do you need to change this?